eventstore/test_views.py

```diff
@@ -387,6 +394,161 @@ def test_postbirth_other_passport_origin(self):
+                    "file": "absolute-filepath-on-coreapp",
+                    "id": "media-id",
+                    "link": "link-to-image-file",
+                    "mime_type": "media-mime-type",
+                    "sha256": "checksum",
+                    "caption": "image-caption",
+                },
+                "location": {
+                    "address": "1 Hacker Way, Menlo Park, CA, 94025",
+                    "name": "location-name",
+                },
+                "system": {"body": "system-message-content"},
+                "text": {"body": "text-message-content"},
+            },
+        ),
+        self.assertEqual(messages.created_by, user.username)
+        self.assertEqual(messages.recipient_type, "type")
+
+    def test_successful_events_request(self):
+        """
+        Should create a new Events object in the database
+        """
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_messages"))
+        self.client.force_authenticate(user)
+        data = {
+            "statuses": [
+                {
+                    "id": "ABGGFlA5FpafAgo6tHcNmNjXmuSf",
+                    "recipient_id": "16315555555",
+                    "status": "read",
+                    "timestamp": "1518694700",
+                }
+            ]
+        }
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        [events] = Events.objects.all()
+        self.assertEqual(str(events.message_id), "ABGGFlA5FpafAgo6tHcNmNjXmuSf")
+        self.assertEqual(str(events.recipient_id), "16315555555")
+        self.assertEqual(events.status, "read")
+        self.assertEqual(
+            events.timestamp, datetime.datetime(2018, 2, 15, 11, 38, 20, tzinfo=UTC)
+        )
+        self.assertEqual(events.created_by, user.username)
+
+    def test_signature_required(self):
+        """
+        Should see if the signature hook is given,
+        otherwise, return a 401
+        """
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_messages"))
+        self.client.force_authenticate(user)
+        data = {
+            "messages": [
+                {
+                    "id": "9e12d04c-af25-40b6-aa4f-57c72e8e3f91",
+                    "from": "sender-wa-id",
+                    "timestamp": "1518694700",
+                    "type": "image",
+                    "message_direction": "I",
+                    "recipient_type": "type",
+                }
+            ]
+        }
+        response = self.client.post(self.url, data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
```

Comment:
We're using the `force_authentication` method for our tests, which makes testing easy, but we should add a test that doesn't have a token in the querystring, and test that we get an auth errors, and then puts the token of the user in the querystring, and tests that the request works fine, because that's how this API endpoint is going to be used.